### PR TITLE
Package: support gitlab tarball urls

### PR DESCRIPTION
**Pull request by travisstaloch**
Allows the package manager to download gitlab tarballs from urls such as `https://gitlab.com/<namespace>/<project>/-/archive/<sha>/<project>-<sha>.tar.gz`

Such http requests have headers `Content-Type=application/octet-stream` and `Content-Disposition='attachment; filename="<project>-<sha>.tar.gz"'`. The package manager doesn't yet support these headers. This patch doesn't attempt to properly parse the content-disposition header. Instead it does a minimal check for the presence of the strings 'attachment' and '.tar.gz'.

I've left the Content-Disposition header parsing to a follow up PR as I'm not sure where it should go.

### CODE REVIEW

File: src/Package.zig [hunk 1]
```diff
@@ -510,6 +510,18 @@ fn fetchAndUnpack(
...
             try unpackTarball(gpa, &req, tmp_directory.handle, std.compress.xz);
+        } else if (ascii.eqlIgnoreCase(content_type, "application/octet-stream")) {
+            // TODO more properly parse the content-disposition header according to https://www.ietf.org/rfc/rfc2183.txt.
+            //      the following is a minimal check as i'm not sure where content-disposition parsing should happen.
+            // support gitlab tarball urls such as https://gitlab.com/<namespace>/<project>/-/archive/<sha>/<project>-<sha>.tar.gz
+            // whose content-disposition header is: 'attachment; filename="<project>-<sha>.tar.gz"'
+            const content_disposition = req.response.headers.getFirstValue("Content-Disposition") orelse
+                return report.fail(dep.url_tok, "Missing 'Content-Disposition' header for Content-Type=application/octet-stream", .{});
+            if (ascii.indexOfIgnoreCasePos(content_disposition, 0, "attachment") != null and
+                ascii.indexOfIgnoreCasePos(content_disposition, 0, ".tar.gz") != null)
```
Comment (andrewrk) [inline src/Package.zig:521]: Use mem.startsWith, mem.endsWith, lose the TODO comment, and this is good to go.

Comment (travisstaloch) [reply src/Package.zig]: Thanks! Done.

**Comment by LordMZTE** [comment]
This seems to have broken gitea/forgejo tarball URLs (which I think is a loss overall :P). My forgejo server seems to cause this particular error:

```
/home/lordmzte/dev/dotfiles/mzte-nv/build.zig.zon:7:20: error: Unsupported 'Content-Disposition' header value: 'attachment; filename="znvim-0a55798f24e084faef68e93f180683157a417639.tar.gz"; filename*=UTF-8''znvim-0a55798f24e084faef68e93f180683157a417639.tar.gz' for Content-Type=application/octet-stream
            .url = "https://mzte.de/git/LordMZTE/znvim/archive/0a55798f24e084faef68e93f180683157a417639.tar.gz",
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I think the additional `filename*=UTF8.....` might be causing problems.

**Comment by DraagrenKirneh** [comment]
Also the PR which most likely "broke" this was not this one, but https://github.com/ziglang/zig/pull/15428 as it changed the check on how to unpack from the ending of the URL path (to better handle redirects) to content-type instead.

**Pull request merged**